### PR TITLE
[shopsys] added Symfony Toolbar to JavaScript Storefront

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, the reason for the PR| It is hard to debug frontend API when running into an error on JavaScript Storefront without Symfony Toolbar and this PR solves that!
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

Screenshot at Jan 04 17-51-58

----
:globe_with_meridians: Live Preview:
  - https://tl-symfony-toolbar.odin.shopsys.cloud
  - https://cz.tl-symfony-toolbar.odin.shopsys.cloud

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -117,6 +117,7 @@
         "league/flysystem-google-cloud-storage": "^3.0",
         "league/iso3166": "^4.2.1",
         "litipk/php-bignumbers": "^0.8.6",
+        "mgdsoft/symfony-profiler-spa": "^1.0",
```
Comment (grossmannmartin) [inline composer.json:120]: I think this should be as a dev dependency and also added in the project-base's composer.json file, right?

File: symfony.lock [hunk 2]
```diff
@@ -330,6 +330,9 @@
     "litipk/php-bignumbers": {
         "version": "0.8.6"
     },
+    "mgdsoft/symfony-profiler-spa": {
+        "version": "v1.0.4"
+    },
```
Comment (grossmannmartin) [inline symfony.lock:335]: can you please add this to the project-base/symfony.lock file?

File: project-base/storefront/components/Basic/SymfonyToolbar/SymfonyToolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import Script from 'next/script';
+
+export const SymfonyToolbar: FC = () => {
+    const { url } = useDomainConfig();
+
+    if (!isEnvironment('development')) {
+        return null;
+    }
+
+    return <Script src={url + '/bundles/symfonyprofilerspa/load_toolbar_spa.js'} strategy="beforeInteractive" />;
```
Comment (grossmannmartin) [inline project-base/storefront/components/Basic/SymfonyToolbar/SymfonyToolbar.tsx:12]:
```suggestion
    return <Script src='/bundles/symfonyprofilerspa/load_toolbar_spa.js' strategy="beforeInteractive" />;
```

should be enough

**Review by grossmannmartin — CHANGES_REQUESTED** [review]
Thanks for the PR 👍  

I'm unsure whether this will be counterproductive – the toolbar will not show all requests, only those made on the client side while browsing the storefront
see missing hashes in the toolbar, that are present in the profiler.

Snímek obrazovky 2024-01-04 v 19 36 40


Snímek obrazovky 2024-01-04 v 19 36 25


Also, after the page refresh the last profile is displayed (for example `privacyPolicyArticle`), but many more queries were executed until the page was rendered. This may be confusing and lead developers to false beliefs that their backend is blazing fast although one of the executed queries is really slow.

File: project-base/storefront/helpers/errors/logException.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import { captureException } from '@sentry/nextjs';
+import { isEnvironment } from 'helpers/isEnvironment';
 
 export const logException = (e: unknown): void => {
-    if (process.env.APP_ENV === 'development') {
+    if (isEnvironment('development')) {
```
Comment (sebaholesz) [inline project-base/storefront/helpers/errors/logException.ts:5]: Thinking if this shouldn't be a separate commit.

File: project-base/storefront/components/Basic/SymfonyToolbar/SymfonyToolbar.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { isEnvironment } from 'helpers/isEnvironment';
+import { useDomainConfig } from 'hooks/useDomainConfig';
+import Script from 'next/script';
+
+export const SymfonyToolbar: FC = () => {
+    const { url } = useDomainConfig();
+
+    if (!isEnvironment('development')) {
```
Comment (sebaholesz) [inline project-base/storefront/components/Basic/SymfonyToolbar/SymfonyToolbar.tsx:8]: How do I turn this feature off as a developer? I say it should be possible to completely turn off (by ENV or whatever). I am not sure about other devs, but I personally do not want it there 😆

**Review by sebaholesz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**